backend/composer/api/serializers.py

```diff
@@ -465,7 +465,20 @@ def get_available_transitions(self, instance) -> list[CSState]:
     def get_has_notes(self, instance) -> bool:
         return instance.has_notes
 
-    def get_statement_preview(self, instance) -> str:
+    def get_journey(self, instance):
```

Comment:
@afonsobspinto this works but I suggest next time to create a separate serializer for the list view. For example we can create a "base" CS serializer without the journey and preview fields. Then use (extend) this one for the list serializer (no extra fields) and extend it (with journey and preview fields) for the get view. Then in the views.py we can pick the serializer per method (list or get). This way it's more clear what the "interface" per api is. For now we can leave it as is.

Reply:
@ddelpiano can you create a tech debt card for this? It's both cleaner and might help a bit in the performance if the calculation of the additional fields is 'expensive'

Reply:
[d4fd9f9](https://github.com/MetaCell/sckan-composer/pull/189/commits/d4fd9f9c210c3536eb835774d44ea18965f8cd34)

Reply:
This actually solves the performance issue on the list view as well. There's probably some attribute besides the journey that is slow to compute. Maybe the from_entities of both vias and destinations